galsim/correlatednoise.py

```diff
@@ -667,6 +667,55 @@ def drawImage(self, image=None, scale=None, wcs=None, dtype=None, wmult=1., add_
             add_to_image=add_to_image, use_true_center=False)
 
+    def drawKImage(self, re=None, im=None, nx=None, ny=None, scale=None, dtype=None, wmult=1.,
+                   add_to_image=False, dk=None):
+        """A method for drawing profiles storing correlation functions (i.e., power spectra) in
+        Fourier space.
+
+        This is a mild reimplementation of the drawKImage() method for GSObjects.  The `gain` is
+        automatically set to unity and cannot be change.  Also, not all the normal parameters of the
+        GSObject method are available.
+
+        If `scale` is not set, and `re` and `im` have no `wcs` attributes, then this will use the
+        wcs of the CorrelatedNoise object.
+
+        @param re           If provided, this will be the real part of the k-space image.  If `re`
+                            and `im` are None, then automatically-sized images will be created.  If
+                            they are given, but the bounds are undefined, then they will be resized
+                            appropriately base on the profile's size. [default: None]
+        @param im           If provided, this will be the imaginary part of the k-space image.  A
+                            provided `im` must match the size and scale of `re`.  If `im` is None,
+                            then `re` must also be None. [default: None]
+        @param scale        If provided, use this as the pixel scale, dk, for the images.  If
+                            `scale` is None and `re` and `im` are given, then take the provided
+                            images' pixel scale (which must be equal).  If `scale` is None and `re`
+                            and `im` are None, then use the Nyquist scale.
+                            If `scale <= 0` (regardless of `re`, `im`), then use the Nyquist scale.
+                            [default: None]
+        @param dtype        The data type to use for automatically constructed images.  Only valid
+                            if `re` and `im` are None. [default: None, which means to use
+                            numpy.float32]
+        @param gain         The number of photons per ADU ("analog to digital units", the units of
+                            the numbers output from a CCD).  [default: 1.]
+        @param wmult        A multiplicative factor by which to enlarge (in each direction) the size
+                            of the image, if you are having drawKImage() automatically construct the
+                            images for you.  [default: 1]
+        @param add_to_image Whether to add to the existing images rather than clear out anything in
+                            the image before drawing.  Note: This requires that `re` and `im` be
+                            provided and that they have defined bounds. [default: False]
+
+        @returns the tuple of Image instances, `(re, im)` (created if necessary)
+        """
+        # Check for obsolete dk parameter
```

Comment:
No need to include the check for the deprecated dk parameter here.  It's a new function. :)